File: src/main/java/maven/example/com/receiving/system/Clipboard.java
package maven.example.com.receiving.system;


import maven.example.com.tools.data.Data;
import maven.example.com.tools.data.TypeData;
import maven.example.com.receiving.tools.Receipt;

import java.awt.*;
import java.awt.datatransfer.DataFlavor;
import java.awt.datatransfer.UnsupportedFlavorException;
import java.io.IOException;

public class Clipboard extends Receipt {
    private static final TypeData TYPE = TypeData.SYSTEM_CLIPBOARD;

    public Clipboard() {
        super(TYPE);
    }

    private String getClipboardText() {
        try {
            return (String) Toolkit.getDefaultToolkit().getSystemClipboard().getData(DataFlavor.stringFlavor);
        } catch (HeadlessException | UnsupportedFlavorException | IOException e) {
            e.printStackTrace();
        }

        return "";
    }

    @Override
    public Data getData() {
        return new Data(TYPE, getClipboardText());
    }
}

File: src/main/java/maven/example/com/receiving/system/Clipboard.kt
package maven.example.com.receiving.system

import maven.example.com.utility.data.Receipt
import maven.example.com.utility.data.Data
import maven.example.com.utility.data.TypeData
import java.awt.HeadlessException
import java.awt.Toolkit
import java.awt.datatransfer.DataFlavor
import java.awt.datatransfer.UnsupportedFlavorException
import java.io.IOException

class Clipboard() : Receipt(TYPE) {

    override val data: Data?
        get() = Data(TYPE, this.clipboardText)

    private val clipboardText: String?
        get() {
            try {
                if (Toolkit.getDefaultToolkit().systemClipboard.isDataFlavorAvailable(DataFlavor.stringFlavor)) {
                    Toolkit.getDefaultToolkit().systemClipboard.getData(DataFlavor.stringFlavor) as String
                } else {
                    null // В буфере не текст (например изображение)
                }
            } catch (e: Exception) {
                System.err.println("Error in clipboard: ${e.localizedMessage}")
                e.printStackTrace()
            }

            return ""
        }

    companion object {
        private val TYPE = TypeData.SYSTEM_CLIPBOARD
    }
}
